### PR TITLE
Paralleltopologytests

### PR DESCRIPTION
**Pull request by LouieVoit**
- Removing a test that tested nothing in IndexConversionTest (since this class is meant to disappear, I don't think it's worth it to even try to update the test)
- Removing some warning in topology classes that were triggered when `getProcessCoordinates` was called with a rank below the topology offset or above the topology offset plus the number of processes for this md simulation. This function is actually called often with an out of bound rank with the MacroscopicCellServiceMacroOnly. So I don't think there should be a warning.
- Finally, adding test for parallel topology classes

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -374,6 +374,10 @@ elseif("${MD_SIM}" STREQUAL "LS1_MARDYN")
     target_link_libraries(mamico INTERFACE adios2::adios2)
   endif()
 endif()
+if(ENABLE_COVERAGE)
+  target_compile_options(mamico INTERFACE --coverage)
```
Comment (Thinkpiet) [inline CMakeLists.txt:378]: Why did you remove `-O0` ? Isn't that needed for coverage?

File: test/unit/coupling/paralleltopology/XYZTopologyTest.cpp [hunk 2]
```diff
@@ -17,8 +18,75 @@ class XYZTopologyTest : public CppUnit::TestFixture
...
+   void testGetProcessCoordinates3D() {
+    unsigned int nx = 2;
+    unsigned int ny = 3;
+    unsigned int nz = 5;
+    tarch::la::Vector<3, unsigned int> numberOfProcesses{nx, ny, nz};
+    unsigned int topologyOffset = 10;
+    coupling::paralleltopology::XYZTopology<3> topology{numberOfProcesses, topologyOffset};
+    for (unsigned int rank = topologyOffset; rank < topologyOffset + nx*ny*nz; ++rank) {
+      tarch::la::Vector<3, unsigned int> processCoordinates = topology.getProcessCoordinates(rank);
+      unsigned int z = (rank-topologyOffset)/(nx*ny);
```
Comment (Thinkpiet) [inline test/unit/coupling/paralleltopology/XYZTopologyTest.cpp:30]: This is correct, but the comment in [ZYXTopology.h](https://github.com/HSU-HPC/MaMiCo/pull/35/files#diff-e0252921926fb82d01c894932d53950a1db66466b5a8138eb99893edaf8c3ebc) line 23 is wrong, so now there is a mismatch between what the XYZTopology unit test is doing and the documentation / the description about what XYZTopology is supposed to do.

I think @jafariv had confused the two files and put the comment for  XYZTopology in the ZYXTopology.cpp file and vice versa. @LouieVoit can you please double-check and correct the comments (in XYZTopology.cpp and ZYXTopology.cpp, lines ca 21 - 26 in both files) before merging this PR? Otherwise the new unit tests are very confusing.

**Review by Thinkpiet — CHANGES_REQUESTED** [review]
In general very nice, just the comments in the class files should be fixed.

File: CMakeLists.txt (hunk 1 above)
Comment (LouieVoit) [reply CMakeLists.txt:378]: There is already -Og in Debug mode. Based on the documentation, it should be enough.

File: test/unit/coupling/paralleltopology/XYZTopologyTest.cpp (hunk 2 above)
Comment (LouieVoit) [reply test/unit/coupling/paralleltopology/XYZTopologyTest.cpp:30]: Done

File: CMakeLists.txt (hunk 1 above)
Comment (Thinkpiet) [reply CMakeLists.txt:378]: Ok right

File: coupling/paralleltopology/ZYXTopology.h [hunk 3]
```diff
@@ -17,13 +17,13 @@ template <unsigned int dim> class ZYXTopology;
...
  *  topologyOffset is used for linearized access of multiple MD instances.
- *Derived class from the class ParallelTopology. E.g. assuming
- *ParallelTopologyType = XYZ and there is a cubic domain, splitted into 8
- *sub-domains (2 sub-domains in each dimension). Then the ordring of the MPI
- *processes is: Rank=0 for x=0,y=0,z=0. Rank=1 for x=1,y=0,z=0. Rank=2 for
- *x=0,y=1,z=0. Rank=3 for x=1,y=1,z=0. Rank=4 for x=0,y=0,z=1. Rank=5 for
- *x=1,y=0,z=1. Rank=6 for x=0,y=1,z=1. Rank=7 for x=1,y=1,z=1.
- *	@brief The XYZTopology orders the ranks in z-y-x manner.
+ *  Derived class from the class ParallelTopology. E.g. assuming
+ *  ParallelTopologyType = XYZ and there is a cubic domain, splitted into 8
```
Comment (Thinkpiet) [inline coupling/paralleltopology/ZYXTopology.h:21]: ZYX, not XYZ
